### ISSUE TITLE
Wrong new messages count

### ISSUE THREAD

**Issue by 532910** [matrix-org/synapse#8598]
Michael says this is synapse not a client ones
https://github.com/vector-im/element-web/issues/15505
https://github.com/vector-im/element-android/issues/2273

> Nope, the count is calculated by the server; `notification_count` https://matrix.org/docs/spec/client_server/r0.6.1#get-matrix-client-r0-sync

_Originally posted by @t3chguy in https://github.com/vector-im/element-web/issues/15505#issuecomment-712769977_

It looks like missed calls are counted multiple times:
Web:
![image](https://user-images.githubusercontent.com/14617699/96571058-c0625000-12d3-11eb-8b31-b958912dbd5f.png)
![image](https://user-images.githubusercontent.com/14617699/96571598-70d05400-12d4-11eb-9106-f2ff96470187.png)

Android:
![image](https://user-images.githubusercontent.com/14617699/96571083-cc4e1200-12d3-11eb-9b1c-78e9486b66eb.png)
![image](https://user-images.githubusercontent.com/14617699/96571464-45e60000-12d4-11eb-8ea4-05114028897b.png)

With `Show hidden events` there 5 events:
![image](https://user-images.githubusercontent.com/14617699/96577332-1fc45e00-12dc-11eb-8a07-c7f2197fee74.png)

Synapse 1.20.1 on debian stable

**Comment by anoadragon453** [comment]
In an encrypted room, the server is unable to tell whether an event is a call, a message, or otherwise, hence we get this behaviour.

Do you experience this problem in unencrypted rooms as well?

**Comment by anoadragon453** [comment]
https://github.com/matrix-org/synapse/blob/dd11f575a29b59aced6cfa7ea7b9faea6f968f8d/synapse/push/baserules.py#L456-L474

**Comment by t3chguy** [comment]
So this is probably a compound issue which should be resolved by the FTUE Notifications project which has a workaround for the server's over-count

**Comment by 532910** [comment]
Yep, this is an encrypted room. For unencrypted it shows correct one event for a missed call.

**Comment by anoadragon453** [comment]
Guessing this is blocked by the notifications project then. Follow https://github.com/matrix-org/matrix-doc/pull/2785 for updates.

**Comment by ajax313** [comment]
in android when you take missed call the event was 3 missed call.

**Comment by erikjohnston** [comment]
I don't think there is anything we on the Synapse team can do here. This behaviour is not ideal, but needs spec work to fix properly.

**Comment by 532910** [comment]
Element team says this is on the synapse side, Synapse team says behaviour is not ideal. Awesome!

**Comment by erikjohnston** [comment]
To be clear: both teams are saying that this requires changes to the spec. That work can be tracked at e.g. https://github.com/matrix-org/matrix-doc/pull/2785. Once any relevant MSC has changed then the Synapse team will do the work to implement it.

Having issues open across lots of different repos makes it much harder to track these issues, and make it way more likely for the work to be forgotten.

**Comment by ArneNostitz** [comment]
I am experiencing the wrong count in all my rooms. The notification is off by 5. Opening and closing the app (element) corrects it though. Element iOS told me this is a server issue. So I found this. Is this the same issue?

**Comment by clokep** [comment]
> I am experiencing the wrong count in all my rooms. The notification is off by 5. Opening and closing the app (element) corrects it though. Element iOS told me this is a server issue. So I found this. Is this the same issue?

Can you cross-link the issue with Element iOS?

**Comment by ArneNostitz** [comment]
i posted first about the issue here https://github.com/vector-im/element-ios/issues/6829. 

on mac i dont see this issue - also reading on Mac adjusts the notification count on iOS ... so its working correctly in the background sending the read receipt through ... but the iOS count is still 5 off .... i still think this is an iOS issue ...

**Comment by ArneNostitz** [comment]
I opened a new issue with a detailed image of what is happening https://github.com/vector-im/element-ios/issues/7135
